### PR TITLE
Lazyload with intersection api with gatsby latest

### PR DESCRIPTION
**Pull request by ashraf-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/hooks/use-handle-lazy-load.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        const target = document.querySelector(targetId)
+        const observer = new IntersectionObserver(handleIntersect, options)
+        observer.observe(target)
+    }
+
+    return is_visible || is_large_screen ? (
+        <>{lazyTemplates}</>
+    ) : (
+        <>
+            <span>laoding</span>
```
Comment (yashim-deriv) [inline src/components/hooks/use-handle-lazy-load.tsx:37]:
```suggestion
            <span>loading</span>
```

Comment (yashim-deriv) [reply src/components/hooks/use-handle-lazy-load.tsx:37]: Probably would need designer help to improve this loader

File: src/pages/home/_our_platforms.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ const OurPlatforms = () => {
             height="fit-content"
             tabletL={{ height: 'fit-content', pb: 'unset' }}
             padding="8rem 0 0 0"
+            id="platform-fold"
```
Comment (yashim-deriv) [inline src/pages/home/_our_platforms.tsx:17]: This id seems to be unused

**Review by yashim-deriv — COMMENTED** [review]
We can move forward with this. Need @prince-deriv  @habib-deriv  reviews as well

**Review by habib-deriv — COMMENTED** [review]
Hi @ashraf-deriv 

1. DP2P section and Sign up for your demo account switched places
2. Also please pull from the latest master, some of the content and design changed.
3. Found the alt of the images for Trade types are missing. Looks like it's production issue as well.
4. Instead of using dummy in the alt of of the dummy images, we can remove the word dummy

**Comment by ashraf-deriv** [comment]
> Hi @ashraf-deriv
> 
> 1. DP2P section and Sign up for your demo account switched places
> 2. Also please pull from the latest master, some of the content and design changed.
> 3. Found the alt of the images for Trade types are missing. Looks like it's production issue as well.
> 4. Instead of using dummy in the alt of of the dummy images, we can remove the word dummy

Hi @habib-deriv , Incorporated your suggestions.

File: src/components/hooks/use-handle-lazy-load.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        const target = document.querySelector(targetId)
+        const observer = new IntersectionObserver(handleIntersect, options)
+        observer.observe(target)
+    }
+
+    return is_visible || is_large_screen ? (
+        <>{lazyTemplates}</>
+    ) : (
+        <>
+            <span>loading</span>
```
Comment (habib-deriv) [inline src/components/hooks/use-handle-lazy-load.tsx:37]: We need to do something with this loading text. It looks bad . Maybe need some loader or something

**Review by habib-deriv — DISMISSED** [review]
LGTM

**Comment by ali-hosseini-fs** [comment]
in the test link there's a horizontal scroll bar on initial render that needs to be fixed

**Comment by ashraf-deriv** [comment]
> in the test link there's a horizontal scroll bar on initial render that needs to be fixed

@ali-hosseini-fs It is fixed.

File: src/components/hooks/use-handle-lazy-load.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            setIsVisible(entry.isIntersecting)
+            observer.disconnect()
+        }
+    }
+    useEffect(() => {
+        createObserver()
+        if (window.innerHeight >= 990) {
+            setLargeScreen(true)
+        }
+    }, [])
```
Comment (ali-hosseini-fs) [inline src/components/hooks/use-handle-lazy-load.tsx:20]: please do the cleanup in the return statement of it 
and also please make sure that it doesn't throw any warning regarding empty dependency array

**Review by ali-hosseini-fs — APPROVED** [review]

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request closed without merging**